benchmarks/Microsoft.AspNetCore.SignalR.Microbenchmarks/DefaultHubDispatcherBenchmark.cs

```diff
@@ -0,0 +1,200 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO.Pipelines;
+using System.Reactive.Linq;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using Microsoft.AspNetCore.Protocols;
+using Microsoft.AspNetCore.SignalR.Internal;
+using Microsoft.AspNetCore.SignalR.Internal.Encoders;
+using Microsoft.AspNetCore.SignalR.Internal.Protocol;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Moq;
+using DefaultConnectionContext = Microsoft.AspNetCore.Sockets.DefaultConnectionContext;
+
+namespace Microsoft.AspNetCore.SignalR.Microbenchmarks
+{
+    public class DefaultHubDispatcherBenchmark
+    {
+        private DefaultHubDispatcher<TestHub> _dispatcher;
+        private HubConnectionContext _connectionContext;
+
+        [GlobalSetup]
+        public void GlobalSetup()
+        {
+            var serviceCollection = new ServiceCollection();
+            serviceCollection.AddSignalRCore();
+
+            var provider = serviceCollection.BuildServiceProvider();
+
+            var serviceScopeFactory = provider.GetService<IServiceScopeFactory>();
+
+            _dispatcher = new DefaultHubDispatcher<TestHub>(
+                serviceScopeFactory,
+                new HubContext<TestHub>(new DefaultHubLifetimeManager<TestHub>()),
+                new Logger<DefaultHubDispatcher<TestHub>>(NullLoggerFactory.Instance));
+
+            var options = new PipeOptions();
+            var pair = DuplexPipe.CreateConnectionPair(options, options);
+            var connection = new DefaultConnectionContext(Guid.NewGuid().ToString(), pair.Transport, pair.Application);
+
+            _connectionContext = new NoErrorHubConnectionContext(connection, TimeSpan.Zero, NullLoggerFactory.Instance);
+
+            var hubProtocolMock = new Mock<IHubProtocol>();
```

Comment:
I wouldn't use Moq in a benchmark. In theory it should return empty method implementations by default but it might call into Moq infrastructure to record invocations etc.